elearning-frontend/src/components/Register.js

```diff
@@ -0,0 +1,195 @@
+        avatarRef.current.value = null;
+        setAvatar(null);
+        setProfilepic('https://st3.depositphotos.com/4111759/13425/v/600/depositphotos_134255710-stock-illustration-avatar-vector-male-profile-gray.jpg')
+    }
+
+    return (
+        <div className="register-card w-3/5 mx-auto">
+            <div className="mb-5">
+                <h4 className='title text-center'>REGISTER</h4>
+            </div>
+            <hr className='mb-5'/>
+            <div>
+                {/* Form */}
+                <form onSubmit={saveUser} encType="multipart/form-data">
+                    <div className='grid grid-cols-4 gap-5'>
+                        <div className="col-span-1">
+                            {/* Avatar */}
+                            <div className="form-group mb-8 avatar-section">
+                                <img src={profilepic} alt="dp" className='mb-5' />
+                                <label>Profile Picture</label>
+                                <div className='grid grid-cols-5 items-center gap-2'>
+                                    <div className='col-span-4'>
+                                        <input
+                                            type="file"
+                                            name="avatar"
+                                            ref={ avatarRef }
+                                            onChange={e => {
+                                                setAvatar(e.target.files[0])
+                                                setProfilepic(URL.createObjectURL(e.target.files[0]))
+                                            }}
+                                            className="appearance-none w-full py-1 px-3 text-gray-700 leading-tight focus:outline-none"
+                                            accept="image/*"
+                                            multiple={ false } />
+                                    </div>
+                                    <div>
+                                        <a
+                                            role="button"
+                                            onClick = { deleteAvatar }
+                                            className="bg-red-600 hover:bg-red-500 py-1 px-2 text-white">X</a>
+                                    </div>
+                                </div>
+                            </div>
+                        </div>
+                        <div className="col-span-3">
+                            <div className='grid grid-cols-3 gap-5'>
+                                {/* First Name */}
+                                <div className="form-group mb-8">
+                                    <label>First Name</label>
+                                    <input type="text" name="firstname" onChange={e => setFirstname(e.target.value)} value={firstname} className="appearance-none border-b-2 w-full py-1 px-3 text-gray-700 leading-tight focus:outline-none" />
```

Comment:
```suggestion
                                    <input type="text" name="firstname" onChange={e => setProfile({...profile, firstName: e.target.value})} value={firstname} className="appearance-none border-b-2 w-full py-1 px-3 text-gray-700 leading-tight focus:outline-none" />
```

So this is how you set, you need to spread first the old data then replace or add the firstname